server/paymentProviders/transferwise/webhook.ts

```diff
@@ -70,12 +75,29 @@ export async function handleTransferStateChange(event: TransferStateChangeEvent)
       platformFeeInHostCurrency: number;
     };
 
-    const paymentOption = expense.data.paymentOption as QuoteV2PaymentOption | QuoteV3PaymentOption;
+    let paymentOption = expense.data.paymentOption as QuoteV2PaymentOption | QuoteV3PaymentOption;
+    // Fetch up-to-date quote to check if payment option has changed
+    const quote = await getQuote(connectedAccount, transfer.quoteUuid);
+    assert(quote, 'Failed to fetch quote from Wise');
+    const wisePaymentOption = quote.paymentOptions.find(p => p.payIn === 'BALANCE' && p.payOut === quote.payOut);
+    if (
+      // Check if existing quote is QuoteV3
+      'price' in paymentOption &&
+      // Check if the priceDecisionReferenceId has changed
+      paymentOption.price.priceDecisionReferenceId !== wisePaymentOption.price?.priceDecisionReferenceId
+    ) {
+      logger.warn(`Wise updated the payment option for expense ${expense.id}, updating existing values...`);
+      const expenseDataQuote = { ...omit(quote, ['paymentOptions']), paymentOption } as ExpenseDataQuoteV3;
+      await expense.update({ data: { ...expense.data, quote: expenseDataQuote, paymentOption } });
+      paymentOption = wisePaymentOption;
```

Comment:
Sounds like we want to reverse these two lines, otherwise the `paymentOption` will not be updated

```suggestion
      paymentOption = wisePaymentOption;
      await expense.update({ data: { ...expense.data, quote: expenseDataQuote, paymentOption } });
```

Reply:
🙌🏼